x-pack/plugins/security_solution/public/timelines/components/timeline/event_details.tsx

```diff
@@ -11,39 +11,105 @@
  */
 
 import { EuiSpacer } from '@elastic/eui';
-import React from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import deepEqual from 'fast-deep-equal';
+import { find } from 'lodash/fp';
 
+import { useDispatch } from 'react-redux';
 import { BrowserFields, DocValueFields } from '../../../common/containers/source';
 import {
   ExpandableEvent,
   ExpandableEventTitle,
+  OnEventDetailsClose,
 } from '../../../timelines/components/timeline/expandable_event';
 import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { TimelineEventsDetailsItem } from '../../../../common/search_strategy/timeline';
+import { useTimelineEventsDetails } from '../../containers/details';
+import { TimelineId } from '../../../../common/types/timeline';
+import { activeTimeline } from '../../containers/active_timeline_context';
+import { timelineActions } from '../../store/timeline';
 
 interface EventDetailsProps {
   browserFields: BrowserFields;
   docValueFields: DocValueFields[];
   timelineId: string;
+  onEventDetailsClose?: OnEventDetailsClose;
 }
 
 const EventDetailsComponent: React.FC<EventDetailsProps> = ({
   browserFields,
   docValueFields,
   timelineId,
+  onEventDetailsClose,
 }) => {
-  const expandedEvent = useDeepEqualSelector(
-    (state) => state.timeline.timelineById[timelineId]?.expandedEvent
+  const dispatch = useDispatch();
+
+  const { expandedEvent, isSaving } = useDeepEqualSelector(
+    (state) => state.timeline.timelineById[timelineId]
```

Comment:
we should use memoized selector + `pick` to avoid unnecessary re-renders like
https://github.com/angorayc/kibana/blob/alerts-details/x-pack/plugins/security_solution/public/timelines/components/flyout/header/index.tsx#L126